samples/shields/x_nucleo_iks02a1/microphone/prj.conf

```diff
@@ -0,0 +1,17 @@
+CONFIG_LOG=n
+CONFIG_BOOT_BANNER=n
+CONFIG_STDOUT_CONSOLE=y
+CONFIG_PRINTK=y
+CONFIG_I2S=y
+CONFIG_GPIO=y
+CONFIG_AUDIO=y
+CONFIG_AUDIO_DMIC=y
+CONFIG_AUDIO_MPXXDTYY=y
+
+# When on-board microphone is used the DMA must run w/o things
+# that could slow it down. So DMA logging must be completely
+# disabled and DMA interrupts must run at maximum priority.
+
+CONFIG_DMA=y
+CONFIG_DMA_0_IRQ_PRI=0
```

Comment:
Can't this be configured in dt ?

Reply:
This would require a change in the stm32 dma driver and related dts. Actually it is a  bit risky to do it here, as it impacts many applications and would require an exhaustive test.

Reply:
ack

Reply:
Well, it was necessary sometime ago, when I did the same for ArgonKey board. Without this definition the audio stream was broken, as the DMA IRQ was not served fast enough. Now I'm relooking again at it and it seems instead that the priority is taken from DTS and it is fixed to 0 by default:

#define DT_ST_STM32_DMA_40026400_IRQ_0_PRIORITY     0

I tried to remove CONFIG_DMA_0_IRQ_PRI from prj.conf and it still works.
I'm not sure where the default priority is set in the dts and how to enforce it.
Any idea? @erwango

Reply:
OK, i think that I found it, sorry.
It is in ./dts/arm/st/f4/stm32f4.dtsi :

interrupts = <56 3 57 3 58 3 59 3 60 3 68 3 69 3 70 3>;

Reply:
Moreover I noticed that CONFIG_DMA and HEAP_SIZE are already set in soc. So I'm gonna delete completely this section.